src/pages/Admin/Venue/VenueDeleteModal.tsx

```diff
@@ -1,106 +1,64 @@
-import React, { useCallback, useState } from "react";
+import React from "react";
 import { Modal } from "react-bootstrap";
-import { useHistory } from "react-router-dom";
+import { useAsyncFn } from "react-use";
 import firebase from "firebase/app";
 
-import { AnyVenue } from "types/venues";
-
-import { WithId } from "utils/id";
+import { ButtonNG } from "components/atoms/ButtonNG";
 
 import "./VenueDeleteModal.scss";
 
-interface PropsType {
-  venue: WithId<AnyVenue>;
+interface VenueDeleteModalProps {
+  venueId?: string;
+  venueName: string;
   show: boolean;
-  onHide: () => void;
+  onHide?: () => void;
+  onDelete?: () => void;
+  onCancel: () => void;
 }
 
-const VenueDeleteModal: React.FunctionComponent<PropsType> = ({
-  venue,
+const VenueDeleteModal: React.FunctionComponent<VenueDeleteModalProps> = ({
+  venueId,
+  venueName,
   show,
   onHide,
+  onDelete,
+  onCancel,
 }) => {
-  const [deleting, setDeleting] = useState(false);
-  const [deleted, setDeleted] = useState(false);
-  const [error, setError] = useState<string>();
-  const history = useHistory();
-
-  const closeDeleteModal = () => {
-    if (deleted) {
-      history.push("/admin");
-    } else {
-      onHide();
-      setDeleted(false);
-    }
-  };
-
-  const deleteVenue = useCallback(async () => {
-    setDeleting(true);
-    try {
-      await firebase.functions().httpsCallable("venue-deleteVenue")({
-        id: venue.id,
-      });
-      setDeleted(true);
-    } catch (error) {
-      setError(error.message);
-    } finally {
-      setDeleting(false);
-    }
-  }, [venue.id]);
+  const [{ error, loading: isDeleting }, deleteVenue] = useAsyncFn(async () => {
+    await firebase.functions().httpsCallable("venue-deleteVenue")({
+      id: venueId,
+    });
+    onDelete?.();
+  }, [onDelete, venueId]);
 
   return (
-    <Modal show={show} onHide={closeDeleteModal}>
-      <div className="venue-delete-modal-container">
-        <h2 className="centered">Delete venue</h2>
+    <Modal show={show} onHide={onHide}>
+      <div className="VenueDeleteModal">
+        <h2 className="centered">Delete space</h2>
         <div className="secondary-action">
-          WARNING: Permanently removes a venue from SparkleVerse
+          WARNING: This action cannot be undone and will permanently remove the
+          space <b>{venueName}</b>!
         </div>
-        {!deleted && (
-          <>
-            <div className="input-group">
-              <span className="info">
-                WARNING: This action cannot be undone! Are you sure you wish to
-                delete {venue.name}?
-              </span>
+        {isDeleting && (
+          <div className="centered-flex">
+            <div className="spinner-border">
+              <span className="sr-only">Loading...</span>
```

Comment:
[OOS] Few of these `Loading...` divs can probably be extracted into existing or new component